File: frontend/src/api/scenarios.ts
import client from "./client";

export interface ScenarioSummary {
  name: string;
  label: string;
  description: string;
  category: string;
  expected_outcome: string;
}

export interface ScenarioDetail extends ScenarioSummary {
  patient_persona: string;
  evaluation_criteria: string[];
  tool_overrides: Record<string, unknown>;
}

export async function listScenarios(): Promise<ScenarioSummary[]> {
  const res = await client.get<ScenarioSummary[]>("/scenarios");
  return res.data;
}

export async function getScenario(name: string): Promise<ScenarioDetail> {
  const res = await client.get<ScenarioDetail>(`/scenarios/${name}`);
  return res.data;
}

File: backend/routers/scenarios.py
"""API routes for scenario definitions."""

from fastapi import APIRouter, HTTPException
from scenarios.definitions import get_scenario, list_scenarios
from schemas.scenario import ScenarioResponse, ScenarioSummary

router = APIRouter(prefix="/api/scenarios", tags=["scenarios"])


@router.get("", response_model=list[ScenarioSummary])
async def list_all_scenarios() -> list[ScenarioSummary]:
    """List all available scenario definitions."""
    return [
        ScenarioSummary(
            name=s.name,
            label=s.label,
            description=s.description,
            category=s.category,
            expected_outcome=s.expected_outcome,
            evaluation_criteria=list(s.evaluation_criteria),
        )
        for s in list_scenarios()
    ]


@router.get("/{scenario_name}", response_model=ScenarioResponse)
async def get_scenario_detail(scenario_name: str) -> ScenarioResponse:
    """Get full details for a specific scenario."""
    scenario = get_scenario(scenario_name)
    if not scenario:
        raise HTTPException(status_code=404, detail="Scenario not found")
    return ScenarioResponse(
        name=scenario.name,
        label=scenario.label,
        description=scenario.description,
        category=scenario.category,
        patient_persona=scenario.patient_persona,
        expected_outcome=scenario.expected_outcome,
        evaluation_criteria=scenario.evaluation_criteria,
        tool_overrides=scenario.tool_overrides,
    )
